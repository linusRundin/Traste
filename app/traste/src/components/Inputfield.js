import React from 'react';
import PropTypes from 'prop-types';
import {TextField} from '@mui/material';
import {Colors} from '../assets/Colors';

/**
 * Custom TextField used for different data in the RenderPage.
 * @param {*} label Name of field.
 * @param {*} value Starting value.
 * @param {*} onChange Handler.
 * @param {*} error What will be shown for the error.
 * @param {*} type What type of values. Usally number or text.
 * @param {*} sx Styling.
 * @return {TextField} Rendered field.
 */
function Inputfield({label, value, onChange, error, type, sx}) {
  return (
    <TextField
      label={label}
      variant='outlined'
      fullWidth
      sx={sx}
      value={value}
      type={type}
      onChange={onChange}
      inputProps={{'data-testid': 'inputfield'}}
      error={!!error}
    />
  );
}

Inputfield.defaultProps = {
  error: false,
  type: 'text',
  sx: {'& .MuiOutlinedInput-root': {
    '&.Mui-focused fieldset': {
      borderColor: Colors.trasteTeal}},
  'marginTop': '15px',
  'backgroundColor': 'rgba(255,255,255,0.3)',
  },
};

Inputfield.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.any.isRequired,
  onChange: PropTypes.func.isRequired,
  error: PropTypes.any.isRequired,
  type: PropTypes.string.isRequired,
  sx: PropTypes.object,
};

export default Inputfield;
