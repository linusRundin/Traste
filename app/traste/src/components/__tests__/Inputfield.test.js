import {render, screen, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Inputfield from '../Inputfield';
import React from 'react';


afterEach(() => {
  cleanup();
});

/**
   * Test function.
   */
function testOnChange() {
  console.log('Change was made');
};

test('Should render inputfield', () => {
  const propsData = {label: 'Docket No.', onChange: testOnChange,
    value: 'string', error: 'error'};

  render(<Inputfield {...propsData} />);

  const inputElement = screen.getByTestId('inputfield');
  expect(inputElement).toBeInTheDocument();
  expect(inputElement).toHaveAccessibleName('Docket No.');
});

test('Should render inputfield', () => {
  const propsData = {label: 'Docket No.', onChange: testOnChange,
    value: 'string', error: 'error'};
  render(<Inputfield {...propsData} />);

  const inputElement = screen.getByTestId('inputfield');
  expect(inputElement).toBeInTheDocument();
  expect(inputElement).not.toHaveAccessibleName('DocketNo.');
});

test('Matches snapshot', () => {
  const propsData = {label: 'Docket No.', onChange: testOnChange,
    value: 'string', error: 'error'};

  const tree = renderer.create(<Inputfield {...propsData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
