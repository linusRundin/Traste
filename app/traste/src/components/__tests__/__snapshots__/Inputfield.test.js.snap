// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1kjrtrh-MuiFormControl-root-MuiTextField-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={true}
  >
    Docket No.
  </label>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={true}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      data-testid="inputfield"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="string"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-1z7n62"
      >
        <span>
          Docket No.
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
